framework/OutStreams/PlotInterfaces/OptParallelCoordinate.py

```diff
@@ -0,0 +1,179 @@
+    """
+      Init of Base class
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'OptParallelCoordinate Plot'
+    self.source = None      # reference to DataObject source
+    self.sourceName = None  # name of DataObject source
+    self.vars = None        # variables to plot
+    self.index = None       # index ID for each batch
+
+
+  def handleInput(self, spec):
+    """
+      Loads the input specs for this object.
+      @ In, spec, InputData.ParameterInput, input specifications
+      @ Out, None
+    """
+    super().handleInput(spec)
+    params, notFound = spec.findNodesAndExtractValues(['source','vars','index'])
+
+    for node in notFound:
+      self.raiseAnError(IOError, "Missing " +str(node) +" node in the OptParallelCoordinatePlot " + str(self.name))
+    else:
+      self.sourceName = params['source']
+      self.vars       = params['vars']
+      self.index      = params['index']
+
+
+  def initialize(self, stepEntities):
+    """
+      Function to initialize the OutStream. It basically looks for the "data"
+      object and links it to the system.
+      @ In, stepEntities, dict, contains all the Objects are going to be used in the
+                                current step. The sources are searched into this.
+      @ Out, None
+    """
+    src = self.findSource(self.sourceName, stepEntities)
+    if src is None:
+      self.raiseAnError(IOError, f'No source named "{self.sourceName}" was found in the Step for SamplePlot "{self.name}"!')
+    self.source = src
+    dataVars = self.source.getVars()
+    missing = [var for var in (self.vars) if var not in dataVars]
+    if missing:
+      msg = f'Source DataObject "{self.source.name}" is missing the following variables ' +\
+            f'expected by OptPath plotter "{self.name}": '
+      msg += ', '.join(f'"{m}"' for m in missing)
+      self.raiseAnError(IOError, msg)
+
+  def run(self):
+    """
+      Main run method
+      @ In, None
+      @ Out, None
+    """
+    data = self.source.asDataset().to_dataframe()
+    ynames  = self.source.getVars(subset='input')
+
+    minGen = int(min(data[self.index]))
+    maxGen = int(max(data[self.index]))
+
+    yMin = np.zeros(4)
+    yMax = np.zeros(4)
+
+    for idx,inp in enumerate(ynames):
+      yMin[idx] = min(data[inp])
+      yMax[idx] = max(data[inp])
+
+    filesID = []
+
+    for idx,genID in enumerate(range(minGen,maxGen+1,1)):
+      population = data[data[self.index]==genID]
+      ys = population[ynames].values
+      fileID = f'{self.name}' + str(genID) + '.png'
+      generateParallelPlot(ys,genID,yMin,yMax,ynames,fileID)
+      filesID.append(fileID)
+
+    fig = plt.figure()
+    with imageio.get_writer(f'{self.name}.gif', mode='I') as writer:
+      for filename in filesID:
+        image = imageio.imread(filename)
+        writer.append_data(image)
+
+
+def generateParallelPlot(zs, batchID, ymins, ymaxs, ynames, fileID):
+  """
+    Main run method.
```

Comment:
Might need to update docstring desc.

For the future generalization, I think this could be abstracted commonly for the Optimizer plots (at least) and then called by the specific algos depending on whether it's using populations or trajectories.

Reply:
This method has been moved to plotUtils so that it can be used by other optimizers